samza-core/src/main/java/org/apache/samza/clustermanager/ContainerManager.java

```diff
@@ -97,7 +152,10 @@ void handleContainerLaunch(SamzaResourceRequest request, String preferredHost, S
    */
   void handleContainerStop(String processorId, String containerId, String preferredHost, int exitStatus,
       Duration preferredHostRetryDelay, ContainerAllocator containerAllocator) {
-    if (standbyContainerManager.isPresent()) {
+    if (hasActiveContainerPlacementAction(processorId)) {
+      LOG.info("Setting the for container state with processorId {} to be stopped because of existing container placement action", processorId);
+      getPlacementActionMetadata(processorId).get().setContainerStatus(ContainerPlacementMetadata.ContainerStatus.STOPPED);
```

Comment:
is there any scenario where getPlacementActionMetadata(processorId).get() fails?

Reply:
The metadata for a previously completed action is not deleted until a new action is accepted, new is only accepted when an existing action is complete, so this should not fail.